spec/components/works/add_files_component_spec.rb

```diff
@@ -3,12 +3,35 @@
 
 require 'rails_helper'
 
+# rubocop:disable RSpec/VerifiedDoubles
+# rubocop:disable RSpec/AnyInstance
 RSpec.describe Works::AddFilesComponent do
   let(:form) { ActionView::Helpers::FormBuilder.new(nil, nil, controller.view_context, {}) }
   let(:rendered) { render_inline(described_class.new(form: form)) }
 
-  it 'renders the component' do
-    expect(rendered.to_html).to include('Add your files')
-    expect(rendered.css('button.dz-clickable').to_html).to include('Choose files')
+  before do
+    allow_any_instance_of(ActionView::Helpers::FormBuilder).to receive(:object).and_return(object)
+  end
+
+  context 'with an unpersisted file component' do
+    let(:object) { double('anonymous form instance', persisted?: false) }
+
+    it 'renders the component' do
+      expect(rendered.to_html).to include('Add your files')
+      expect(rendered.css('button.dz-clickable').to_html).to include('Choose files')
+    end
+  end
+
+  context 'with a persisted file component' do
+    let(:model) { create(:attached_file, :with_file) }
+    let(:object) { double('anonymous form instance', persisted?: true, model: model) }
```

Comment:
```suggestion
  let(:form) { ActionView::Helpers::FormBuilder.new(nil, work_form, controller.view_context, {}) }
  let(:rendered) { render_inline(described_class.new(form: form)) }
  let(:work_form) { WorkForm.new(object) }

  context 'with an unpersisted file component' do
    let(:object) { build(:work) }

    it 'renders the component' do
      expect(rendered.to_html).to include('Add your files')
      expect(rendered.css('button.dz-clickable').to_html).to include('Choose files')
    end
  end

  context 'with a persisted file component' do
    let(:model) { create(:attached_file, :with_file) }
    let(:object) { build(:work, attached_files: [model]) }
```

Reply:
How about this?

Reply:
Looks good. Looks like the trick is injecting a form as the "object" of the form builder? I'll patch it in. Thanks!

Reply:
yes.

Reply:
@jcoyne pushed. have another look?